### PR TITLE
fixes #24055 - docker search unauth repos

### PR DESCRIPTION
**Pull request by thomasmckay**
To test, similar to https://github.com/Katello/katello/pull/7459 but _docker search_ should find unauth'ed repos when logged in or not logged in.

### CODE REVIEW

**Comment by thomasmckay** [comment]
@akofink - follow-on PR for container images, if you have time

**Comment by akofink** [comment]
@thomasmckay you've got some failures here

File: app/controllers/katello/api/registry/registry_proxies_controller.rb [hunk 1]
```diff
@@ -48,7 +49,7 @@ def optional_authorize
 
     def registry_authorize
       @repository = find_readable_repository
-      return true if request.method == 'GET' && @repository && @repository.environment.registry_unauthenticated_pull
+      return true if request.method == 'GET' && @repository && @repository.registry_unauthenticated_pull
```
Comment (akofink) [inline app/controllers/katello/api/registry/registry_proxies_controller.rb]: Is this an intentional change? This column doesn't exist, and there's no delegation to the env.

**Comment by thomasmckay** [comment]
@akofink - bad rebase fixed

**Comment by thomasmckay** [comment]
[test katello]

**Pull request merged**